backend/service/k8s/k8s.go

```diff
@@ -38,9 +39,13 @@ func New(cfg *any.Any, logger *zap.Logger, scope tally.Scope) (service.Service,
 				Precedence: c.Kubeconfigs,
 			}
 		}
+
+		if c.RestClientConfig != nil {
+			restClientConfig = *c.RestClientConfig
+		}
 	}
 
-	c, err := newClientsetManager(loadingRules, logger)
+	c, err := newClientsetManager(loadingRules, restClientConfig, logger)
```

Comment:
pass in `c.RestClientConfig` and add a nil check to the apply func instead.

then we don't have to do the nil check, assignment dance in every caller if we want to reuse func.